pkg/logs/input/file/tailer.go

```diff
@@ -76,7 +76,13 @@ func NewDecoderFromSourceWithPattern(source *config.LogSource, multiLinePattern
 		lineParser = parser.KubernetesFormat
 		matcher = &decoder.NewLineMatcher{}
 	case config.DockerSourceType:
-		lineParser = parser.DockerFileFormat
+		switch coreConfig.Datadog.GetBool("logs_config.use_podman_logs") {
+		case false:
+			lineParser = parser.DockerFileFormat
+		case true:
+			// podman's on-disk logs are in kubernetes format
+			lineParser = parser.KubernetesFormat
+		}
```

Comment:
nit: I'm a bit curious about use of a `switch` here for a boolean value. Any reason why we wouldn't use an `if`/`else` or just an `if`?

```go
lineParser = parser.DockerFileFormat
if coreConfig.Datadog.GetBool("logs_config.use_podman_logs") {
    lineParser = parser.KubernetesFormat
}
```

Reply:
I think I was just trying to keep with the nested-switch structure.  But, you're right, it's weird :)